### PR TITLE
07-flex-layout-2: Add gap to container class instead of adding margin to card class in solution.css

### PR DESCRIPTION
**Pull request by coder09dev**
## Because

In `solution.css`, adding `gap` property to the parent `.container` class, instead of adding margin to the child `.card` class since we are using `Flexbox` for creating the cards. 

## This PR

- Removing `margin: 16px`  from the `.card` class.
- Instead, adding `gap: 16px` property to the `.container` class to give space between the cards.

## Issue

Closes #XXXXX

### Output

Screenshot 2024-03-08 at 7 27 19 PM

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `01-flex-center: Update self check`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If applicable, I have ensured that the TOP solution files match the Desired Outcome image

### CODE REVIEW

File: flex/07-flex-layout-2/solution/solution.css [hunk 1]
```diff
@@ -73,11 +73,11 @@ a {
   padding: 32px;
   display: flex;
   flex-wrap: wrap;
+  gap: 16px;
```
Comment (MaoShizhong) [inline flex/07-flex-layout-2/solution/solution.css:76]:
```suggestion
  gap: 32px;
```
To match the original desired outcome image, you'd need a gap of 32px as the previous 16px margin would be doubled (no margin collapse).

File: flex/07-flex-layout-2/solution/solution.css [hunk 2]
```diff
@@ -73,11 +73,11 @@ a {
   padding: 32px;
```
Comment (MaoShizhong) [inline flex/07-flex-layout-2/solution/solution.css:73]:
```suggestion
  padding: 48px;
```
The container padding will also need increasing to account for the lack of "outer" 16px spacing if no child margin.

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
@coder09dev Thanks for opening this PR. I agree that it's a little more semantic to space flex items via `gap` rather than `margin` in many cases nowadays.

Could you make the following changes to make sure that the solution still produces a perfect match with the desired outcome image?
Simply replacing the child margin with a parent gap would not result in an identical image.

File: flex/07-flex-layout-2/solution/solution.css (hunk 1 above)
Comment (coder09dev) [reply flex/07-flex-layout-2/solution/solution.css]: updated gap to 32px.

File: flex/07-flex-layout-2/solution/solution.css (hunk 2 above)
Comment (coder09dev) [reply flex/07-flex-layout-2/solution/solution.css]: increased padding to `48px`.

**Review by MaoShizhong — APPROVED** [review]
Many thanks!

**Pull request merged**